template-for-repository/proofs/Makefile.common

```diff
@@ -115,16 +115,34 @@ CBMCFLAGS += --unwind 1 $(CBMC_UNWINDSET)
 ifneq ($(strip $(EXTERNAL_SAT_SOLVER)),)
    CHECKFLAGS += --external-sat-solver $(EXTERNAL_SAT_SOLVER)
 endif
-CHECKFLAGS += --bounds-check
-CHECKFLAGS += --conversion-check
-CHECKFLAGS += --div-by-zero-check
-CHECKFLAGS += --float-overflow-check
-CHECKFLAGS += --nan-check
-CHECKFLAGS += --pointer-check
-CHECKFLAGS += --pointer-overflow-check
-CHECKFLAGS += --signed-overflow-check
-CHECKFLAGS += --undefined-shift-check
-CHECKFLAGS += --unsigned-overflow-check
+
```

Comment:
I thought you and Kareem had a conversation saying that this could not work.

If this does work, there needs to be some documentation here telling the user that these are the flags we recommend, and how to omit flags that the proof project decides not to use.

Reply:
So, the discussion was that having this following in `Makefile.common`

```
MALLOC_CAN_FAIL=--malloc-fail-null
```

and this in (harness-specific) `Makefile`:

```
undefine MALLOC_CAN_FAIL
```

won't work. This is a different solution, described by @karkhaz in a comment above.

Adding some documentation is a good idea. Should I add it here, or add an example within `template-for-proof`, or both?

Reply:
I added some description above these variables. Please take a look.